src/gluonts/dataset/split/splitter.py

```diff
@@ -198,61 +182,69 @@ def _train_slice(self, item: TimeSeriesSlice) -> TimeSeriesSlice:
+            return item[-max_history + label_len :]
         else:
             return item
 
-    def split(self, items: List[DataEntry]) -> TrainTestSplit:
-        split = TrainTestSplit()
+    def split(self, dataset: Dataset):
+        test_data = TestTemplate(dataset=dataset, splitter=self)
+        train_data = TrainingDataset(dataset=dataset, splitter=self)
 
-        for item in map(TimeSeriesSlice.from_data_entry, items):
+        return train_data, test_data
 
+    def _generate_train_slices(self, items: List[DataEntry]):
+        for item in map(TimeSeriesSlice.from_data_entry, items):
             train = self._train_slice(item)
-            test = self._trim_history(self._test_slice(item))
-
-            split._add_train_slice(train)
-
-            prediction_length = getattr(self, "prediction_length")
-
-            _check_split_length(train.end, test.end, prediction_length)
-            split._add_test_slice(test)
 
-        return split
+            yield train.to_data_entry()
 
-    def rolling_split(
+    def _generate_test_slices(
         self,
-        items: List[DataEntry],
-        windows: int,
-        distance: Optional[int] = None,
-    ) -> TrainTestSplit:
-        # distance defaults to prediction_length
+        items: Dataset,
+        prediction_length: int,
+        windows,
+        distance,
+        max_history,
+    ):
         if distance is None:
-            distance = getattr(self, "prediction_length")
-        assert distance is not None
-
-        split = TrainTestSplit()
+            distance = prediction_length
 
         for item in map(TimeSeriesSlice.from_data_entry, items):
             train = self._train_slice(item)
-            split._add_train_slice(train)
 
             for window in range(windows):
                 offset = window * distance
-                test = self._trim_history(
-                    self._test_slice(item, offset=offset)
+                test = self._test_slice(
+                    item, prediction_length=prediction_length, offset=offset
+                )
+
+                _check_split_length(
+                    train.end, test[1].end, train.end.freq * prediction_length
                 )
-                prediction_length = getattr(self, "prediction_length")
 
-                _check_split_length(train.end, test.end, prediction_length)
-                split._add_test_slice(test)
+                input = self._trim_history(
+                    test[0], len(test[1]), max_history
+                ).to_data_entry()
+
+                label = test[1].to_data_entry()
+                df_label = pd.DataFrame(label["target"])
+                df_label = df_label.set_index(
+                    pd.period_range(
+                        start=label["start"],
+                        periods=len(label["target"]),
+                        freq=train.end.freq,
+                    )
+                )
 
-        return split
+                yield input, df_label
```

Comment:
I think it's better to `yield label` directly, so that the output here is a pair of `DataEntry` objects. Conversion to `pandas` can happen later wherever needed